trumania/core/util_functions.py

```diff
@@ -89,23 +89,33 @@ def merge_2_dicts(dict1, dict2, value_merge_func=None):
     if dict1 is None:
         return dict2
 
-    def merged_value(key):
-        if key not in dict1:
-            return dict2[key]
-        elif key not in dict2:
-            return dict1[key]
-        else:
-            if value_merge_func is None:
-                raise ValueError(
-                    "Conflict in merged dictionaries: merge function not "
-                    "provided but key {} exists in both dictionaries".format(
-                        key))
-
-            return value_merge_func(dict1[key], dict2[key])
-
-    keys = set(dict1.keys()) | set(dict2.keys())
-
-    return {key: merged_value(key) for key in keys}
+    if dict1 == dict2:
+        for k, v in dict1.items():
+            dict1[k] = value_merge_func(v, v)
```

Comment:
Why do we need to call `value_merge_func()` ? why not just returning `dict1` ? I just did a quick check, python's `==` seems to honor `__eq__()` on the dictionnary values => if `==` returns true, that means those dicts are equal and I think do not need to be merged

Reply:
If the merge function is for example `add 1`, you still need to apply the merge function.